### PR TITLE
Refactoring packages error processing

### PR DESCRIPTION
**Pull request by ghost**
Unifyed approach to form response codes from gitea errors in packages module.

### CODE REVIEW

File: services/packages/packages.go [hunk 1]
```diff
@@ -26,9 +26,9 @@ import (
 )
 
 var (
-	ErrQuotaTypeSize   = errors.New("maximum allowed package type size exceeded")
-	ErrQuotaTotalSize  = errors.New("maximum allowed package storage quota exceeded")
-	ErrQuotaTotalCount = errors.New("maximum allowed package count exceeded")
+	ErrQuotaTypeSize   = util.NewInvalidArgumentErrorf("maximum allowed package type size exceeded")
+	ErrQuotaTotalSize  = util.NewInvalidArgumentErrorf("maximum allowed package storage quota exceeded")
+	ErrQuotaTotalCount = util.NewInvalidArgumentErrorf("maximum allowed package count exceeded")
```
Comment (KN4CK3R) [inline services/packages/packages.go]: Is "you have reached a limit" an "argument error"?

Comment (d1nch8g) [reply services/packages/packages.go:31]: Nope. Was the closest type that was defined in the util package. Would it be better to expand util's error types?

Comment (d1nch8g) [reply services/packages/packages.go:31]: Would error `FailedPrecondition` (by analogy with gRPC status codes) fit this case?

Comment (ghost) [reply services/packages/packages.go]: @KN4CK3R 

No. Added draft with `FailedPrecondition` (similar with gRPC spec) error in [this branch](https://github.com/d1nch8g/gitea/tree/refactoring-package-upload-error-handle-failed-precondition). Would it be valid for current scenario?

Comment (KN4CK3R) [reply services/packages/packages.go]: Is it a mistake that only `ErrQuotaTotalCount` is a `util.NewFailedPreconditionErrorf`?

Comment (d1nch8g) [reply services/packages/packages.go]: My bad, I was not sure about this error, changed to `util.ErrLimitExceeded`

Comment (d1nch8g) [reply services/packages/packages.go]: Yes, applied required corrections

**Comment by d1nch8g** [comment]
@KN4CK3R 

I added some corrections and simplifactions to enhance the error handling process.

1. Added `ErrPayloadTooLarge` and `ErrLimitExceeded` to Gitea errors.
2. Added functions that automatically get response codes from shared Gitea errors.
3. Added variadic syntax for response codes to `apiError(...)` functions in all package types to make the response code an optional parameter (should keep the existing flexible format but make error assignment simpler).
5. Added logging to errors in the container module; unified error format for all container responses.
6. Changed status code for size errors to `413 Payload Too Large` and reached limit errors to `409 Conflict`

I am not sure about this implementation or whether this approach might be applicable. This model might be useful because different internal error changes will be handled automatically, potentially reducing the number of accidental errors, but it requires additional attention to error messages since those messages might be passed to the client side.

**Pull request closed without merging**